cms/djangoapps/contentstore/views/item.py

```diff
@@ -103,6 +103,28 @@ def _filter_entrance_exam_grader(graders):
     return graders
 
 
+def set_discussion_toggle(value, xblock, user):
+    """
+    Recursively update all children of given xblock in modulestore
+    """
+    is_block_with_discussion_enabled = (
+        xblock.location.block_type in ["vertical", "sequential", "chapter", "course"]
+    )
+    if not all([isinstance(value, bool), is_block_with_discussion_enabled]):
+        return
+
+    if hasattr(xblock, "discussion_enabled"):
+        xblock.discussion_enabled = value
+        _update_with_callback(xblock, user)
+    else:
+        for child in xblock.get_children():
+            if child.category in ["chapter", "sequential"]:
+                set_discussion_toggle(value, child, user)
+            elif hasattr(child, "discussion_enabled"):
+                child.discussion_enabled = value
+                _update_with_callback(child, user)
```

Comment:
Why is this needed? If you just call `set_discussion_toggle` it will already check if the block has "discussion_enabled" (only for verticals and courses) and set the value and call the update function. 
```suggestion
            set_discussion_toggle(value, child, user)
```